src/chromatix/elements/convenience.py

```diff
@@ -0,0 +1,21 @@
+import flax.linen as nn
+from chromatix import Field
+from chex import Array
+
+__all__ = ["Flip", "OffsetScale"]
```

Comment:
We should add another one that's called `SampleSpectrum` which should look something like this:

```python
class SampleSpectrum(nn.Module):
    num_wavelength_samples: int

    @nn.compact
    def __call__(self, field: Field, key: KeyArray) -> Field:
        # TODO(dd): this can only be used as the first layer in a sequence for now
        return field.sample_spectrum(key, self.num_wavelength_samples)
```

and we have a method in `Field` that looks like:

```python
def sample_spectrum(
    self,
    key: KeyArray,
    num_wavelengths: int
) -> Field:
    spectrum = jax.random.choice(key, jnp.squeeze(self.spectrum), shape=(num_wavelengths,), replace=False)
    spectral_density = jax.random.choice(key, jnp.squeeze(self.spectral_density), shape=(num_wavelengths,), replace=False)
    spectrum = rearrange(jnp.atleast_1d(spectrum), "c -> 1 1 1 c")
    spectral_density = rearrange(jnp.atleast_1d(spectral_density), "c -> 1 1 1 c")
    spectral_density = spectral_density / jnp.sum(spectral_density)
    return self.replace(
        spectrum=spectrum,
        spectral_density=spectral_density
    )
```